python/py_object.h

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#ifndef SCIPP_PYTHON_PY_OBJECT_H
+#define SCIPP_PYTHON_PY_OBJECT_H
+
+#include "pybind11.h"
+
+namespace py = pybind11;
+
+namespace scipp::python {
+
+/// Wrapper around pybind11::object to provide deep copy and deep comparison.
+class PyObject {
+public:
+  PyObject() = default;
+  PyObject(PyObject &&other) = default;
+  PyObject &operator=(PyObject &&other) = default;
+
+  PyObject(const PyObject &other) { *this = other; }
+  PyObject &operator=(const PyObject &other) {
+    py::object copy = py::module::import("copy");
+    py::object deepcopy = copy.attr("deepcopy");
+    m_object = deepcopy(other.m_object);
+    return *this;
+  }
```

Comment:
This class is doing about what I would expect. This PR proposes no change to documentation, but I think that library users should be aware what is happening to their wrapped python objects and that should be documented in our help.

Reply:
It is documented as part of the `copy` methods, see, e.g., https://scipp.readthedocs.io/en/latest/generated/scipp.DatasetProxy.html#scipp.DatasetProxy.

Do you have further suggestions where a good place for mentioning it would be?

Reply:
At present I don't think we have example documentation covering the `Variable(value=...)` type variable use. I think it would probably be mentioned along side there. At the moment, there would be nothing to do.

Reply:
What you mention is documented in this section: https://scipp.readthedocs.io/en/latest/user-guide/data-structures.html#0-D-variables-(scalars).

Reply:
Maybe I can add an FAQ item?

Reply:
I am now adding a section on "copying" on the "How to" page https://scipp.readthedocs.io/en/latest/user-guide/how_to.html, which will highlight this